pkg/sync/sync_context.go

```diff
@@ -563,8 +563,12 @@ func (sc *syncContext) getSyncTasks() (_ syncTasks, successful bool) {
 
 		if sc.applyOutOfSyncOnly {
 			if modified, ok := sc.modificationResult[k]; !modified && ok {
-				sc.log.WithValues("resource key", k).V(1).Info("Skipping as resource was not modified")
-				continue
+				if sc.prune && resource.Target == nil {
```

Comment:
I think we should not analyze `sc.prune` here, just assume resource is modified if `resource.Target` or `resource.Live` is nil. This way sync result will have information about resource that have to be pruned but skipped since prune is disabled.